### ISSUE TITLE
AbstractAdapter::createApiException() not actual

### ISSUE THREAD

**Issue by PiloTeZ** [php-tmdb/api#118]
Requested API, the API returned an error, caught exception Undefined property: stdClass::$status_code" in class AbstractAdapter on the line 35, due to the fact that stdObject $errors has no key and status_code status_code. 
Apparently the TMDB API has been updated. Attached screenshots.
Query: discover/movie; page=1775

![2016-07-03_14-36-30](https://cloud.githubusercontent.com/assets/3667701/16545191/bcf8578a-412b-11e6-89f0-1f3fa9185672.png)
![2016-07-03_14-37-08](https://cloud.githubusercontent.com/assets/3667701/16545192/bd11233c-412b-11e6-8380-02a28ed21ab3.png)

**Comment by wtfzdotnet** [comment]
You seem to be indexing pages above 1000 for starters, which is documented at tmdb api as the maximum limit.

Nonetheless a good find, I've never came across this before, I'll create a fix for this sometime.

**Comment by PiloTeZ** [comment]
Thanks, good plugin )

**Comment by thinkingmedia** [comment]
I've found that this is a valid bug.

The `AbstractAdapter` assumes that all responses will contain a body of JSON, but this is not the case.

There are times when the API goes down for maintenance and the server responses with a 500 status code and HTML body. In these cases a critical PHP error is raised because `$errors` is `null`.

You can hack the `AbstractAdapter` as follows to work around this issue.

```
    protected function createApiException(Request $request, Response $response)
    {
        $errors = json_decode((string)$response->getBody());

        return new TmdbApiException(
            isset($errors->status_code) ? $errors->status_code : $response->getCode(),
            isset($errors->status_message) ? $errors->status_message : 'invalid response',
            $request,
            $response
        );
    }
```

This resolves the `null` reference error, but I'm wondering if the `TmdbApiException` is the right exception for HTTP status code 500.

What status code you get is unpredictable. I think it depends on the state of what maintenance is happening on the API. I've caught 500, 501 and 502 errors while using the API.

It would be easier as a developer to handle these errors if the API threw a new exception called `ServerErrorException` instead of `TmdbApiException` because these errors are more HTTP protocol related than API related.

This can be fixed in the `GuzzleAdapter`

Can I make a pull request to fix this?

**Comment by wtfzdotnet** [comment]
@thinkingmedia if you have more feedback on the changes in `2.1` please let me know ( and I'd really welcome PR's ).

@PiloTeZ , this fix has been added in both 2.0 and 2.1. However I will discontinue 2.0 from this point forward.
